sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1419,6 +1429,35 @@ private static EventProcessorOptions CreateOptions(EventProcessorClientOptions c
             };
         }
 
+        /// <summary>
+        /// Creates, starts, and enriches processing diagnostics scope.
+        /// </summary>
+        /// <param name="eventData">The instance of <see cref="EventData"/> which is being processed.</param>
+        /// <returns>The instance of scope.</returns>
+        private DiagnosticScope StartProcessorScope(EventData eventData)
+        {
+            var diagnosticScope = ClientDiagnostics.CreateScope(DiagnosticProperty.EventProcessorProcessingActivityName, ActivityKind.Consumer, MessagingDiagnosticOperation.Process);
+            if (diagnosticScope.IsEnabled)
```

Comment:
```suggestion

            if (diagnosticScope.IsEnabled)
```

Should we just do an early exit here to reduce the nesting?

```csharp
if (!diagnosticScope.IsEnabled)
{
    return;
}
```